lib/util/cache/repository/impl/repository-cache-impl.ts

```diff
@@ -1,43 +1,35 @@
 import { promisify } from 'util';
 import zlib from 'zlib';
 import hasha from 'hasha';
-import upath from 'upath';
-import { GlobalConfig } from '../../../../config/global';
+import type { RepositoryCacheType } from '../../../../config/types';
 import { logger } from '../../../../logger';
-import { outputCacheFile, readCacheFile } from '../../../fs';
 import {
   CACHE_REVISION,
   isValidRev10,
   isValidRev11,
   isValidRev12,
 } from '../common';
-import type { RepoCacheRecord } from '../types';
-import { RepoCacheBase } from './base';
+import type { CacheClient, RepoCache, RepoCacheData } from '../types';
+import { CacheClientFactory } from './cache-client-factory';
 
 const compress = promisify(zlib.brotliCompress);
 const decompress = promisify(zlib.brotliDecompress);
 
-export class LocalRepoCache extends RepoCacheBase {
+export class RepositoryCacheImpl implements RepoCache {
```

Comment:
```suggestion
export class RepositoryCacheBase implements RepoCache {
```

this should be the base class for all cache implementations (beside null cache), it should have 2 abstract protected methods for `_load` and `_save` which is called from this class

Reply:
@viceice 
This is what I intend to implement, hope this is what you meant
![image](https://user-images.githubusercontent.com/97394622/184840084-0da4578b-324a-47c6-850a-67ebea393650.png)

_load and _save through me off quite a bit as these are cache methods, while the clients have read/write

Reply:
sounds good, function name was a sample, read and write is better. 👍